waspc/data/Generator/templates/server/src/_types/index.ts

```diff
@@ -20,24 +22,40 @@ export type AuthenticatedAction<Entities extends _Entity[], Input, Output> =
 
 type AuthenticatedOperation<Entities extends _Entity[], Input, Output> = (
   args: Input,
-  context: Expand<OperationContext<Entities> & { 
-  // TODO: This type must match the logic in core/auth.js (if we remove the
-  // password field from the object there, we must do the same here). Ideally,
-  // these two things would live in the same place:
-  // https://github.com/wasp-lang/wasp/issues/965
-    {= userFieldName =}: Omit<{= userEntityName =}, 'password'> 
-  }>,
+  context: ContextWithUser<Entities>,
 ) => Promise<Output>
 
+export type AuthenticatedApi<
+  Entities extends _Entity[],
+  Params extends ExpressParams,
+  ResBody,
+  ReqBody,
+  ReqQuery extends ExpressQuery,
+  Locals extends Record<string, any>
+> = (
+  req: Request<Params, ResBody, ReqBody, ReqQuery, Locals>,
+  res: Response<ResBody, Locals>,
+  context: ContextWithUser<Entities>,
+) => void
+
 {=/ isAuthEnabled =}
 type Operation<Entities extends _Entity[], Input, Output> = (
   args: Input,
-  context: Expand<OperationContext<Entities>>,
+  context: Context<Entities>,
 ) => Promise<Output>
 
-type OperationContext<Entities extends _Entity[]> = {
-  entities: Expand<EntityMap<Entities>>
-}
+export type Api<
+  Entities extends _Entity[],
+  Params extends ExpressParams,
+  ResBody,
+  ReqBody,
+  ReqQuery extends ExpressQuery,
+  Locals extends Record<string, any>
+> = (
+  req: Request<Params, ResBody, ReqBody, ReqQuery, Locals>,
+  res: Response<ResBody, Locals>,
+  context: Context<Entities>,
```

Comment:
I see you decided to share the context between operations and the API after all.

Fine by me, just making sure you didn't leave it by accident (as the reasoning was that we maybe wouldn't want to accidentally change one while changing the other)..

Reply:
Yeah I decided to share for now and we can separate if/when they diverge. Just wanted to keep it simple for now 👍🏻